### PR TITLE
Accelerated datasets can fallback to federated source while loading

### LINKED ISSUE #3250: Accelerated datasets can fallback to federated source if not ready
Add a config option on acceleration parameters: `fallback_on_load` (or something) that allows the dataset to fallback to the federated source while the initial acceleration load is happening.

Also mark the dataset as "ready" for the readiness endpoint immediately if this is set.

### PR DESCRIPTION
**Pull request by phillipleblanc**
## 🗣 Description

This PR introduces a new option to the control the behavior of accelerated tables during their initial data load. The previous behavior was to have the `/v1/ready` API return `Not Ready` until all datasets were fully loaded. This is still the default behavior with `ready_state: on_load`. However, now when a query is executed against an accelerated table that is still loading data configured with `ready_state: on_load`, an error is returned indicating the acceleration is still initializing. Previously this would have returned zero results.

A new option `ready_state: on_registration` has been added to allow the `/v1/ready` API to return `Ready` immediately after the accelerated table is registered, and any queries on the accelerated table during the initial load to fallback to the federated source directly.

Example Spicepod:

```yaml
datasets:
  - from: s3://my_bucket/my_dataset/
    name: my_dataset
    params:
      ...
    acceleration:
      enabled: true
      ready_state: on_registration # Will mark the dataset as ready immediately and queries on this table will fallback to the S3 source directly until the initial acceleration is complete.
  - from: s3://my_bucket/my_dataset2/
    name: my_dataset2
    params:
      ...
    acceleration:
      enabled: true
      ready_state: on_load # Will mark the dataset as ready only after the initial acceleration is complete. Queries against the dataset will return an error before the load has completed.
```

The most important changes include adding the `ReadyState` enum, updating the `AcceleratedTable` and `Refresher` structs, and modifying related methods to incorporate this new behavior.

## 🔨 Related Issues

Closes #3250
Part of #3185

## 📄 Documentation Requirements

This will require docs changes to document the new acceleration behavior.

## 🤔 Concerns

It might make sense to have the accelerated table always query the federated source while the initial load is in progress, and have this new option only affect the readiness API.

### CODE REVIEW

**Review by sgrebnov — DISMISSED** [review]

**Comment by Jeadie** [comment]
> It might make sense to have the accelerated table always query the federated source while the initial load is in progress, and have this new option only affect the readiness API.

I think this is reasonable.

File: bin/spiced/src/lib.rs [hunk 1]
```diff
@@ -226,7 +226,9 @@ pub async fn run(args: Args) -> Result<()> {
 
     let cloned_rt = rt.clone();
     let server_thread =
-        tokio::spawn(async move { cloned_rt.start_servers(args.runtime, tls_config).await });
+        tokio::spawn(
```
Comment (Jeadie) [inline bin/spiced/src/lib.rs:229]: Why did we have to pin this now?

**Review by Jeadie — DISMISSED** [review]

File: bin/spiced/src/lib.rs (hunk 1 above)
Comment (phillipleblanc) [reply bin/spiced/src/lib.rs:229]: I was getting this lint error: https://rust-lang.github.io/rust-clippy/master/index.html#large_futures

Comment (Jeadie) [reply bin/spiced/src/lib.rs:229]: I just got it too 😆

**Review by sgrebnov — APPROVED** [review]

**Pull request merged**